client/jetpack-connect/authorize.js

```diff
@@ -274,15 +275,15 @@ export class JetpackAuthorize extends Component {
 	}
 
 	shouldRedirectJetpackStart( props = this.props ) {
-		const { partnerSlug } = props;
-		const partnerRedirectFlag = config.isEnabled(
+		const { partnerSlug, partnerID } = props;
+		const pressableRedirectFlag = config.isEnabled(
 			'jetpack/connect-redirect-pressable-credential-approval'
 		);
 
 		// If the redirect flag is set, then we conditionally redirect the Pressable client to
 		// a credential approval screen. Otherwise, we need to redirect all other partners back
 		// to wp-admin.
-		return partnerRedirectFlag ? partnerSlug && 'pressable' !== partnerSlug : partnerSlug;
+		return pressableRedirectFlag ? partnerID && 'pressable' !== partnerSlug : partnerID;
```

Comment:
This logic feels off - if `partnerId` is defined, use `partnerSlug`. Did you mean to just prepend `partnerId &&` to the existing logic? Should we use either the ID or the slug only?

Also, to make it more readable, should we translate the logic to multiple lines?

Reply:
In this context, `partnerID` is a signal that the `partner_id` query param is set in the URL, which we can then use to make an assumption that the connection attempt is from the Jetpack Start flow. The `partnerSlug` is set depending on whether that `partnerID` value is whitelisted in Calypso to a certain partner.

So, to solve the reported issue, while also not breaking Pressable functionality, we need to check whether `partnerID` is set, and if so, then check if the `partnerSlug` is Pressable or not.

I've changed the logic so that it's a bit more readable, at least in my opinion. 😄